src/org/zaproxy/zap/model/SessionStructure.java

```diff
@@ -269,11 +286,14 @@ private static RecordStructure addStructure (Session session, String host, HttpM
 					tmpMsg = getTempHttpMessage(session, parentName, msg);
 					parentHistoryId = tmpMsg.getHistoryRef().getHistoryId();
 				}
-				RecordStructure parentRs = addStructure(session, host, tmpMsg, paths, size-1, parentHistoryId); 
+				RecordStructure parentRs = addStructure(session, host, tmpMsg, paths, size-1, parentHistoryId, false); 
 				parentId = parentRs.getStructureId();
 			}
 			msgRs = Model.getSingleton().getDb().getTableStructure().insert(
 					session.getSessionId(), parentId, historyId, nodeName, url, method);
+		} else if (newOnly) {
```

Comment:
That would break the functionality - we should only check 'newOnly' if msgRs is not null